### PR TITLE
Add/help center launchpad navigation v1

### PR DESCRIPTION
**Pull request by agrullon95**
#### Proposed Changes

* Added HelpCenter entry point to Launchpad screen if `launchpad_screen` option is `full`

#### Testing Instructions

* Checkout this branch
* Sandbox public API and a launchpad enabled site
* Navigate to Calypso `apps/editing-toolkit` and execute `yarn dev --sync`
* Navigate to the site-editor and click the question mark icon on the top right navigation bar to open the help center.
* Verify you are able to see the Launchpad entry point and that it navigates to the launchpad screen correctly.
* Test this out on another screen such as the `/posts` page.

image

Hover effect:

image

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #72276

### CODE REVIEW

**Comment by agrullon95** [comment]
cc: @nuriapenya

**Comment by nuriapenya** [comment]
@agrullon95 As we can't add the number of steps yet, could we make the circle a bit smaller? Otherwise, it looks somehow empty.

I would suggest making it **32x32**. Reference design here: 9wpoYJrMlYKAKUYvVBMN74-fi-2184%3A32375

Screenshot 2023-01-30 at 16 57 12

File: packages/help-center/src/components/help-center-launchpad.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { SITE_STORE } from '../stores';
+
+export const HelpCenterLaunchpad = () => {
+	const { __ } = useI18n();
+
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const site = useSelect( ( select ) => siteId && select( SITE_STORE ).getSite( siteId ) );
+	const siteIntent = site && site?.options?.site_intent;
+	const siteSlug = site && new URL( site.URL ).host;
+	const launchpadURL = `https://wordpress.com/setup/${ siteIntent }/launchpad?siteSlug=${ siteSlug }`;
```
Comment (edanzer) [inline packages/help-center/src/components/help-center-launchpad.tsx]: We've worked through several of our own issues with having links respect the environment we're in (ie, wordpress.com vs calypso.localhost:3000, etc. Not sure if it's worth doing so here via use of relative urls or some other mechanism?

File: packages/help-center/src/components/help-center-launchpad.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	const sectionName = useSelector( ( state ) => getSectionName( state ) );
+
+	const handleLaunchpadHelpLinkClick = () => {
+		recordTracksEvent( 'calypso_help_launchpad_click', {
```
Comment (edanzer) [inline packages/help-center/src/components/help-center-launchpad.tsx:34]: Good job just anticipating and adding the tracks events from the start.

File: packages/help-center/src/components/help-center-launchpad.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	if ( ! site || ! siteIntent || ! siteSlug ) {
+		return null;
+	}
+	return (
+		<div className="inline-help__launchpad-container">
+			<a
+				className="inline-help__launchpad-link"
+				href={ localizeUrl( launchpadURL ) }
+				target="_blank"
```
Comment (edanzer) [inline packages/help-center/src/components/help-center-launchpad.tsx]: Is intentional to have this open in a separate/new tab?

**Review by alshakero — COMMENTED** [review]
Beautiful code. Does this need to work on Atomic sites. Because I think it won't work now due to accessing `site_options`.

File: packages/help-center/src/components/help-center-launchpad.tsx (hunk 3 above)
Comment (agrullon95) [reply packages/help-center/src/components/help-center-launchpad.tsx]: I changed this `target="_top"` and it seems to be able to open the launchpad link within the same window now.

File: packages/help-center/src/components/help-center-launchpad.tsx (hunk 1 above)
Comment (agrullon95) [reply packages/help-center/src/components/help-center-launchpad.tsx]: Yes , we definitely want to add that here. I'm still looking into a way to get the URL pathname to navigate to the correct environment.

**Comment by agrullon95** [comment]
> Beautiful code. Does this need to work on Atomic sites. Because I think it won't work now due to accessing `site_options`.

Yes we want this to work on atomic sites as well. Are you aware of a way to access site options for atomic sites within this ETK/Help center?

**Review by alshakero — APPROVED** [review]
OK it seems we do pass `site_options` to Atomic sites. I just tested the Help Center and it gets this info already.

I encourage you to test on Atomic after deploying, it's a bit of a hassle to test now and it seems your code handles the lack of `site_options` well.

Do you need help deploying ETK?

File: packages/help-center/src/components/help-center-launchpad.tsx (hunk 1 above)
Comment (agrullon95) [reply packages/help-center/src/components/help-center-launchpad.tsx]: I added the logic to handle the redirection for different environments

**Comment by agrullon95** [comment]
@nuriapenya - I updated the Circular progress bar size to 32x32

image

**Review by Addison-Stavlo — APPROVED** [review]
Tests well and looks good on my end. Thanks Anthony!

**Comment by agrullon95** [comment]
> OK it seems we do pass `site_options` to Atomic sites. I just tested the Help Center and it gets this info already.
> 
> I encourage you to test on Atomic after deploying, it's a bit of a hassle to test now and it seems your code handles the lack of `site_options` well.
> 
> Do you need help deploying ETK?

Yes I could definitely use some help deploying ETK when you get a chance

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7788121

Thank you @agrullon95 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7788121).

**Pull request merged**